modules/openapi-generator/src/main/resources/php/ObjectSerializer.mustache

```diff
@@ -278,7 +278,7 @@ class ObjectSerializer
             return $values;
         }
 
-        if (substr($class, 0, 4) === 'map[') { // for associative array e.g. map[string,int]
+        if (preg_match('/^(array<|map\[)/', $class)) { // for associative array e.g. array<string,int>
```

Comment:
Most of the changes are documentation. There could be no code instances since map was not a valid keyword/type. 

This is kinda the glue for what should be the only technical change. When deserializing this handles but the new generated value and the old map value in case it was being references in some code.